persistence/test/test.go

```diff
@@ -22,39 +26,43 @@ type TestDBSettings struct {
 func PGTestDSN() string {
 	dsn, ok := os.LookupEnv("LNMUX_TEST_DB_DSN")
 	if !ok {
-		dsn = "postgres://bottle:bottle@localhost:45432/bottle_test?sslmode=disable"
+		dsn = "postgres://bottle:bottle@localhost:45432/template1?sslmode=disable"
 	}
 
 	return dsn
 }
 
-func ResetPGTestDB(t *testing.T, settings *TestDBSettings) (*pg.DB, string) {
+func CreatePGTestDB(t *testing.T, settings *TestDBSettings) *pg.Options {
 
+	// To create a DB, we need a connection to the server that isn't to
+	// the nonexistent DB itself. We connect to the template1 DB, which
+	// always exists.
 	if settings.DSN == "" {
 		settings.DSN = PGTestDSN()
 	}
-	dsn := settings.DSN
-
-	if settings.Schema == "" {
-		settings.Schema = "lnmux"
-	}
 
 	if settings.ExpectedSchemaVersion == 0 {
 		settings.ExpectedSchemaVersion = PGExpectedSchemaVersion
 	}
 
 	dbSettings, err := pg.ParseURL(settings.DSN)
 	require.NoError(t, err)
-	require.NoError(t, err)
 
-	db := pg.Connect(dbSettings)
+	defaultDb := pg.Connect(dbSettings)
+	dbSettings = defaultDb.Options()
 
-	_, err = db.Exec(fmt.Sprintf("DROP SCHEMA IF EXISTS %s CASCADE", settings.Schema))
-	require.NoError(t, err)
+	randDbName := fmt.Sprintf("bottle_test_%d",
+		atomic.AddUint32(&dbSuffix, 1))
```

Comment:
What is the idea behind starting with a random number and incrementing? Don't we risk left-over dbs on each run still?

Reply:
Good catch, thanks! Fixed.